code/modules/antagonists/slaughter/slaughter.dm

```diff
@@ -259,21 +259,32 @@
 		if(!M)
 			continue
 
+		// Unregister the signal first, otherwise it'll trigger the "ling revived inside us" code
+		UnregisterSignal(M, COMSIG_MOB_STATCHANGE)
+
 		M.forceMove(T)
 		if(M.revive(full_heal = TRUE, admin_revive = TRUE))
 			M.grab_ghost(force = TRUE)
 			playsound(T, feast_sound, 50, TRUE, -1)
 			to_chat(M, "<span class='clown'>You leave [src]'s warm embrace,	and feel ready to take on the world.</span>")
 
 /mob/living/simple_animal/hostile/imp/slaughter/laughter/bloodcrawl_swallow(mob/living/victim)
-	if(consumed_mobs)
-		// Keep their corpse so rescue is possible
-		consumed_mobs += victim
-	else
-		// Be safe and just eject the corpse
-		victim.forceMove(get_turf(victim))
-		victim.exit_blood_effect()
-		victim.visible_message("<span class='warning'>[victim] falls out of the air, covered in blood, looking highly confused. And dead.</span>")
+	// Keep their corpse so rescue is possible
+	consumed_mobs += victim
+	RegisterSignal(victim, COMSIG_MOB_STATCHANGE, .proc/on_victim_statchange)
+
+/mob/living/simple_animal/hostile/imp/slaughter/laughter/proc/on_victim_statchange(mob/victim, new_stat)
+	if(!isliving(victim))
+		return
+
+	var/mob/living/living_victim = victim
+	if(new_stat != DEAD)
```

Comment:
```suggestion
	if(victim.stat != DEAD || new_stat == DEAD)
		return
```
* Early return
* You want to expel them if they transition from dead to living, but not if they fall to crit or w/e (is that even possible?)